### PR TITLE
🔧 upgrade tslint

### PR DESCRIPTION
**Pull request by giladgray**
#### Changes proposed in this pull request:

- upgrade `tslint` and related packages
- fix all the fails (all of them were `--fix`able)

#### Reviewers should focus on:

- does this count as a major version rev of `@blueprintjs/tslint-config`? I don't think so.

### CODE REVIEW

File: packages/datetime/src/dateRangePicker.tsx [hunk 1]
```diff
@@ -144,7 +144,10 @@ export class DateRangePicker extends AbstractPureComponent<IDateRangePickerProps
         [`${SELECTED_RANGE_MODIFIER}-end`]: day => DateUtils.areSameDay(this.state.value[1], day),
 
         [HOVERED_RANGE_MODIFIER]: day => {
-            const { hoverValue, value: [selectedStart, selectedEnd] } = this.state;
+            const {
+                hoverValue,
+                value: [selectedStart, selectedEnd],
+            } = this.state;
```
Comment (giladgray) [inline packages/datetime/src/dateRangePicker.tsx]: 🤷‍♂️

File: packages/core/src/components/slider/handle.tsx [hunk 2]
```diff
@@ -223,8 +223,12 @@ export class Handle extends AbstractPureComponent<IInternalHandleProps, IHandleS
...
 
         const sizeKey = vertical
-            ? useOppositeDimension ? "width" : "height"
-            : useOppositeDimension ? "height" : "width";
+            ? useOppositeDimension
+                ? "width"
+                : "height"
+            : useOppositeDimension
+            ? "height"
+            : "width";
```
Comment (giladgray) [inline packages/core/src/components/slider/handle.tsx]: 🤷‍♂️

File: packages/datetime/src/dateRangePicker.tsx [hunk 3]
```diff
@@ -564,12 +569,12 @@ export class DateRangePicker extends AbstractPureComponent<IDateRangePickerProps
...
-    *   a) the proposed [Month, Year] change throws the two calendars out of order
-    *   b) the proposed [Month, Year] goes beyond the min / max months
-    * and rectify appropriately.
-    */
+     * The min / max months are offset by one because we are showing two months.
+     * We do a comparison check to see if
+     *   a) the proposed [Month, Year] change throws the two calendars out of order
+     *   b) the proposed [Month, Year] goes beyond the min / max months
+     * and rectify appropriately.
+     */
```
Comment (giladgray) [inline packages/datetime/src/dateRangePicker.tsx]: 👌

**Comment by adidahiya** [comment]
no, you don't have to major version rev `@blueprintjs/tslint-config`

**Pull request merged**